From 279ff49a7f73ad2bbce446b3d73c37a36acdb021 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 12 Sep 2024 08:26:55 +0200
Subject: [PATCH 1110/1695] plat-stm32mp2: enable remoteproc driver in optee
 system services

Do not remove remoteproc driver from secure services as the driver must
be probed to allow linux to access some registers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fbdca3dee645193e40eceb367c66b77969269092 plat-stm32mp2: enable remoteproc driver in optee system services]
Change-Id: I102c012d54f1f3cea461c1ea7a26409bfe6ce5f7
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/403915
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/conf.disable-secure-services.mk | 3 ---
 core/arch/arm/plat-stm32mp2/conf.mk                         | 1 -
 2 files changed, 4 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.disable-secure-services.mk b/core/arch/arm/plat-stm32mp2/conf.disable-secure-services.mk
index fe97e8965..e3817f44b 100644
--- a/core/arch/arm/plat-stm32mp2/conf.disable-secure-services.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.disable-secure-services.mk
@@ -13,9 +13,6 @@ $(call force,CFG_TEE_BENCHMARK,n)
 $(call force,CFG_WITH_TUI,n)
 $(call force,CFG_WITH_USER_TA,n)
 
-# Deactivate remote processor support
-$(call force,CFG_STM32MP_REMOTEPROC,n)
-
 # Deactivate crypto support
 $(call force,CFG_CRYPTO,n)
 $(call force,CFG_STM32_CRYP,n)
diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 6d8335c36..27b8dc31f 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -231,7 +231,6 @@ ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_HASH CFG_STM32_PKA \
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
-CFG_STM32MP_REMOTEPROC ?= n
 CFG_DRIVERS_REMOTEPROC ?= $(CFG_STM32MP_REMOTEPROC)
 CFG_REMOTEPROC_PTA ?= $(CFG_STM32MP_REMOTEPROC)
 ifeq ($(CFG_REMOTEPROC_PTA),y)
-- 
2.47.2

